modules/data-streams/src/main/java/org/elasticsearch/datastreams/lifecycle/downsampling/DeleteSourceAndAddDownsampleToDS.java

```diff
@@ -91,8 +95,22 @@ ClusterState execute(ClusterState state) {
                 return ClusterState.builder(state).metadata(newMetaData).build();
             }
         } else {
-            // the source index exists
             DataStream sourceParentDataStream = sourceIndexAbstraction.getParentDataStream();
+            if (sourceParentDataStream != null && sourceParentDataStream.getWriteIndex().getName().equals(sourceBackingIndex)) {
+                String errorMessage = String.format(
+                    Locale.ROOT,
+                    "index [%s] is the write index for data stream [%s] and cannot be replaced",
+                    sourceBackingIndex,
+                    sourceParentDataStream.getName()
+                );
+                throw new IllegalStateException(errorMessage);
+            }
+
+            IndexMetadata sourceIndexMeta = state.metadata().index(sourceBackingIndex);
+            assert sourceIndexMeta != null : "the source index abstraction exists already";
```

Comment:
nit: This assertion could be confusing -- someone would wonder why it's a problem if the source index exists already (i.e. the message seems backwards doesn't it?)

Reply:
Ha, that is a confusing message 😬 Updated it